### ISSUE TITLE
Issue of add widget on notebookPanel.contentHeader

### ISSUE THREAD

**Issue by c3-ashleys** [jupyterlab/jupyterlab#15681]
Hi there,

We recently upgrade Jupyterlab from 3.6.3 to 4.0.9. One issue we found out was that in our extension, we have logic to add a banner widget inside of a notebook. in 3.6 we used `notebookPanel.contentHeader.addWidget(widget)`(widget here is ReactWidget not IWidgetExtension) and it worked fine. after upgrading, we found out that the widget is still added correctly and we were able to retrieve the element in the DOM, however it's no longer visually visible(with proper css). Is this intended in 4? is it a bug?

<img width="1792" alt="Screenshot 2024-01-22 at 1 42 34 PM" src="https://github.com/jupyterlab/jupyterlab/assets/85262269/8a2315c8-5246-47bd-af49-e3639c494460">

**Comment by krassowski** [comment]
I would say this is a bug in a sense. I also see that https://github.com/jupyterlab/extension-examples/blob/main/contentheader/README.md was incorrectly updated because it no longer refers to `contentheader`.

**Comment by c3-ashleys** [comment]
is it gonna be a fix in the source code? is there workaround we could add in our extension?

**Comment by krassowski** [comment]
Yes, a workaround that modifies dimensions of the header could work, but maybe it is better to fix it at source straight-away. Let me take a look.

**Comment by krassowski** [comment]
I am no longer sure if this is a bug with nested `BoxPanel`s, but indeed it can be solved with rather simple workaround of setting `.node.style.minHeight` on the widget that you intend to add to the panel, e.g.:

```js
widget.node.style.minHeight = '20px';
main.contentHeader.addWidget(widget);
```

**Comment by c3-ashleys** [comment]
i did try `widget.node.style.height = '100px'` before and it didn't work for me
